homeassistant/components/overkiz/water_heater_entities/hitachi_dhw.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Hitachi DHW."""
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_HIGH_DEMAND,
+    STATE_OFF,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    OverkizCommandParam.STANDARD: STATE_ECO,
+    OverkizCommandParam.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    OverkizCommandParam.STOP: STATE_OFF,
+}
+
+OPERATION_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_OPERATION_MODE.items()}
+
+
+class HitachiDHW(OverkizEntity, WaterHeaterEntity):
+    """Representation of Hitachi DHW."""
+
+    _attr_min_temp = 30.0
+    _attr_max_temp = 70.0
+    _attr_precision = PRECISION_WHOLE
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+    )
+    _attr_operation_list = [*OPERATION_MODE_TO_OVERKIZ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_DHW_TEMPERATURE)
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.MODBUS_CONTROL_DHW_SETTING_TEMPERATURE
+            ),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_CONTROL_DHW_SETTING_TEMPERATURE, int(temperature)
+        )
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation ie. eco, electric, performance, ..."""
+        if (
+            self.executor.select_state(OverkizState.MODBUS_CONTROL_DHW)
```

Comment:
When there is only one state, I prefer to be more explicit:
```suggestion
            self.device.states[OverkizState.MODBUS_CONTROL_DHW]
```

Reply:
Done there and in others places too.

Reply:
Great it will ease the code.